source/application/AMPIC887DriverServer/AMPIC887AxisMap.h

```diff
@@ -0,0 +1,297 @@
+#ifndef AMPIC887AXISMAP_H
+#define AMPIC887AXISMAP_H
+
+#include <QMap>
+#include <QList>
+#include "AMGCS2.h"
+#include "AMPIC887AxisCollection.h"
+#include "AMGCS2Support.h"
+#include "util/AMCArrayHandler.h"
+/*!
+  * A class which maps Axes of a PI C887.11 controller to a type T.
+  */
+template <typename T>
+class AMPIC887AxisMap : public QMap<AMGCS2::Axis, T>
+{
+public:
+	/*!
+	  * Creates an instance of an axis map, which is empty.
+	  */
+	AMPIC887AxisMap() {}
+
+	/*!
+	  * Returns a collection of the axes which this map contains.
+	  */
+	AMPIC887AxisCollection axes() const
+	{
+		AMPIC887AxisCollection returnCollection(AMPIC887AxisCollection::EmptyCollection);
+		QList<AMGCS2::Axis> axisEntries = this->keys();
```

Comment:
Okay, did a bit more research into this.
It turns out that in order to call a base class function, the compiler needs to know which base it is from. In the case of a template with unspecified type T there is no way of determining this. The compiler doesn't yet know the type of T which is specified, and it doesn't assume that it will necessarily be the same as the instantiated template type from which this is called. As such with yet unknown template types you have to specify 'this' so that it knows to call the keys() function for the same template type.

tldr. I can remove the this pointer from all the specialization classes below, but not the top one for type T.